### DISCUSSION TITLE
Directory when opening a new shell

### DISCUSSION THREAD

**Discussion by o6uoq** [marlonrichert/zsh-autocomplete#308] [Q&A] [answered]
Is there a flag/option to configure showing the current directory listing vs not at all but still maintaining auto-complete based on input? E.g. I open a new tab, prompt is as default without the 'Directory' listing below, but if I type out a directory file or folder, it will provide the autocomplete.

**Comment by marlonrichert** [comment, marked as answer]
Add this to your `.zshrc` file:
```zsh
zstyle ':autocomplete:*' min-input 1
```

<details>
<summary>(Old answer)</summary>

If
* you do _not_ have `.` in your path

_and_

* you have _unset_ the [`AUTO_CD` shell option](https://zsh.sourceforge.io/Doc/Release/Options.html#index-AUTOCD),

then local dirs & executables will not be shown at an empty prompt.

Failing that, you can override the [`_autocomplete.config.tag-order` function](https://github.com/marlonrichert/zsh-autocomplete/blob/79534a25eb6a956b32f860032e98445b41193004/module/.autocomplete.config#L37).

</details>

**Reply by marlonrichert** [reply]
@o6uoq There is a simpler solution than I originally suggested. Check my updated answer, above.

**Reply by o6uoq** [reply]
> @o6uoq There is a simpler solution than I originally suggested. Check my updated answer, above.

That worked, thank you! Is there documentation for auto-complete settings in terms of what is displayed when we type? E.g. if I type `g` I see auto-complete for aliases, shell functions, builtin commands, et al. but namely parameters, which displays things I do *not* want in my shell like `GITHUB_TOKEN`. Is this configurable?

I'd like auto-complete for some things, but not all things.

**Reply by marlonrichert** [reply]
> I see auto-complete for aliases, shell functions, builtin commands, et al. but namely parameters, which displays things I do not want in my shell like GITHUB_TOKEN.
> I'd like auto-complete for some things, but not all things.

I'm working on making this better as part of #346.

> Is there documentation for auto-complete settings in terms of what is displayed when we type?
> Is this configurable?

Yes, it is configurable, but it's not documented at present. If you want to fiddle with it, try overriding [the `_autocomplete.config.tag-order` function](https://github.com/marlonrichert/zsh-autocomplete/blob/f208d047f30eb2e74e890fa22f88e9f47b0862d9/scripts/.autocomplete.config#L46). However, once I'm done with #346, that might not be necessary anymore.
